### ISSUE TITLE
i18n: Missing singular

### ISSUE THREAD

**Issue by rakekniven** [nextcloud/ios#3248]
https://github.com/nextcloud/ios/blob/bcfb4031eb0b116128378dc1a36b8234613b716e/iOSClient/Supporting%20Files/en.lproj/Localizable.strings#L681

**Comment by i2h3** [comment]
I introduced this in #3204 and it got reverted in #3270 while being reintroduced today in #3311. I have not thought about this. Given the context of the surrounding user interface, this can be removed and the "%d remaining" used instead. This reduces redundant wording and avoids pluralization.

**Comment by Valdnet** [comment]
The string `%d remaining` should also be declined in plural.
It does not need to be declined in plural for the string: `Remaining: %d`.

Example for Polish:
```
Pozostało 1
Pozostały 2
Pozostały 5
```
I suggest returning the text string that was `%d remaining downloads allowed`.
This is a more user-friendly sentence.

**Comment by i2h3** [comment]
This requires the introduction of a [`*.stringsdict`](https://developer.apple.com/documentation/xcode/localizing-strings-that-contain-plurals) resource for pluralized strings in the Xcode project. I am on it already but @tobiasKaminsky needs to set up an additional resource in the Transifex project [like Talk for iOS has it](https://app.transifex.com/nextcloud/nextcloud/talk-ios-plurals/).

**Comment by i2h3** [comment]
@tobiasKaminsky Please see my [feature branch for the updated Transifex configuration](https://github.com/nextcloud/ios/blob/3248-missing-singular/iOSClient/.tx/config#L11). Please verify it because I am not abled to add resources to Transifex. This way it should work but I am not fully sure about the type. It should be correct, according to what I read on Transifex.

**Comment by tobiasKaminsky** [comment]
As Mark had, I had also troubles, thus I removed "zero" plural type.
I am pretty sure, that this is somehow wrong.

But at least it is now available: https://app.transifex.com/nextcloud/nextcloud/ios-plurals/
And you can try to update it.

**Comment by i2h3** [comment]
> another resource? ;-)

@rakekniven Yes, this unfortunately is the way it works in Xcode projects. Localized strings are usually managed in the primitive key-value format of `.strings` files. The more complex logic of pluralization is not feasible in that way and implemented in XML (more specifically: Property Lists) documents known as `.stringsdict` (see [this example in my feature branch](https://github.com/nextcloud/ios/commit/72ff2ab3412ce224347c809340f7ba5d2a48ccda#diff-236f9c27e0fce1ecb4d04424ed27e8cc56c4575effb2a3b57021f033f03a55c4)). The additional resource is inevitable. 😕

Though, there is hope ahead: [string catalogues](https://developer.apple.com/documentation/xcode/localizing-and-varying-text-with-a-string-catalog). They were introduced in 2023 with Xcode 15 which are the new way to manage localization within an Xcode project and only requires a single JSON document for everything, IIRC.

I did refrain to switch the whole project by the way to string catalogs because of one plural because of the potential dozen regressions, nerves of my team members and what work it would cause on Transifex. 😅 

> Can you have a look?

I will. When creating the localization files, the possible cases are added automatically. Apparently Transifex checks for plausibility depending on localized language and does not simply ignore inapplicable cases as I assumed.

> Would it be a solution to only add "one" and "other" and Nextcloud will add the other values depending on the language?

I would have assumed translators on Transifex would do that step because my language competence does not reach _that_ far to know about pluralization in all of our supported languages. 😅 Anyway, it might be the best start to reduce all to what you suggested.

> As Mark had, I had also troubles, thus I removed "zero" plural type.

@tobiasKaminsky Probably language dependent, too, because Transifex lists the zero case explicitly for Arabic. I think Transifex is more strict than necessary here. In English, that case can be used to have a designated "No more downloads allowed" string, just for example.

**Comment by nickvergessen** [comment]
Here is what iOS Talk did to add plural translations:
https://github.com/nextcloud/talk-ios/pull/861

I'd assume it should also work for iOS Files?

**Comment by nickvergessen** [comment]
The resource is there on transifex by now, so merging the `.tx/config` change from https://github.com/nextcloud/ios/commit/72ff2ab3412ce224347c809340f7ba5d2a48ccda into [develop](https://github.com/nextcloud/docker-ci/blob/d12cd00b6725f0d68460adf5fe6be449e84b2b0d/translations/handleiOSTranslations.sh#L12-L14) should be good enough?

**Comment by i2h3** [comment]
@nickvergessen The way Talk for iOS does it is the same implemented in my feature branch. I did not have time yet to update it to resolve the errors mentioned above. Hopefully today.
